### PR TITLE
fix(cluster_k8s): add rack feature

### PR DESCRIPTION
**Pull request by dkropachev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/221

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo/225

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1093,6 +1094,7 @@ def get_installed_packages(self):
             return None
 
     def destroy(self):
+        self.stop_task_threads()
```
Comment (bentsi) [inline sdcm/cluster.py:1097]: need to discuss

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -181,29 +181,37 @@ def publish_event(self, disrupt, status=True, data=None):
...
-            all_nodes = [node for node in self.cluster.nodes if not node.running_nemesis]
-            if dc_idx:
-                all_nodes = [node for node in all_nodes if node.dc_idx == dc_idx]
-            self.target_node = random.choice(all_nodes)
+    def _get_target_nodes(
+            self,
+            is_seed: Optional[bool] = None,
+            dc_idx: Optional[int] = None,
+            rack: Optional[int] = None) -> list:
+        output = [node for node in self.cluster.nodes if not node.running_nemesis]
```
Comment (bentsi) [inline sdcm/nemesis.py:189]: nodes =

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -181,29 +181,37 @@ def publish_event(self, disrupt, status=True, data=None):
...
-                self.log.warning("Cluster doesn't contain free nonseeds nodes. Test will failed")
-            self.target_node = random.choice(non_seed_nodes)
-        else:
-            all_nodes = [node for node in self.cluster.nodes if not node.running_nemesis]
-            if dc_idx:
-                all_nodes = [node for node in all_nodes if node.dc_idx == dc_idx]
-            self.target_node = random.choice(all_nodes)
+    def _get_target_nodes(
+            self,
+            is_seed: Optional[bool] = None,
```
Comment (bentsi) [inline sdcm/nemesis.py:186]: why not bool

File: sdcm/nemesis.py [hunk 4]
```diff
@@ -629,8 +635,8 @@ def disrupt_terminate_and_replace_node(self):  # pylint: disable=invalid-name
         InfoEvent(message="StartEvent - Terminate node and wait 5 minutes").publish()
         self._terminate_cluster_node(self.target_node)
         time.sleep(300)  # Sleeping for 5 mins to let the cluster live with a missing node for a while
-        InfoEvent(message="FinishEvent - target_node was terminated").publish()
-        new_node = self._add_and_init_new_cluster_node(old_node_ip)
+        InfoEvent(message='FinishEvent - target_node was terminated')
```
Comment (bentsi) [inline sdcm/nemesis.py:638]: add back call to `.publish()`

**Review by bentsi — APPROVED** [review]

**Pull request merged**